labels-config-service-impl/src/main/java/org/hypertrace/label/config/service/LabelsConfigServiceImpl.java

```diff
@@ -167,9 +180,10 @@ public void updateLabel(
             responseObserver.onError(new StatusRuntimeException(Status.ALREADY_EXISTS));
             return;
           }
-          configServiceCoordinator.getLabel(requestContext, updatedLabelInReq.getId());
-          Label updatedLabelInRes =
-              configServiceCoordinator.upsertLabel(requestContext, updatedLabelInReq);
+          labelStore
+              .getObject(requestContext, updatedLabelInReq.getId())
+              .orElseThrow(() -> new StatusRuntimeException(Status.NOT_FOUND));
```

Comment:
I guess this answers my question, although this would be a confusing response if upserting a read only label.

Reply:
It is checked first and we throw this error - `responseObserver.onError(new StatusRuntimeException(Status.INVALID_ARGUMENT));`